uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java

```diff
@@ -0,0 +1,274 @@
+            final String joinFinishDialogText = joinFinishDialog.getText();
+            final String joinStatus = joinFinishDialogText.split(":")[1];
+            // The status should be successful
+            Assert.assertTrue("SUCCESSFUL".equalsIgnoreCase(joinStatus));
+
+            // dismiss the dialog
+            UiAutomatorUtils.handleButtonClick("android:id/button1");
+
+            // compare the UPN to make sure joined with the expected account
+            final String joinedUpn = getAccountUpn();
+            Assert.assertTrue(expectedUpn.equalsIgnoreCase(joinedUpn));
+        } catch (final UiObjectNotFoundException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    @Nullable
+    @Override
+    public String obtainDeviceId() {
+        launch(); // launch Broker Host app
+
+        if (shouldHandleFirstRun) {
+            handleFirstRun(); // handle first run experience
+        }
+
+        UiAutomatorUtils.handleButtonClick("com.microsoft.identity.testuserapp:id/buttonDeviceId");
+
+        // Look for the device id dialog box
+        final UiObject deviceIdDialog = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                "android:id/message"
+        );
+
+        Assert.assertTrue(deviceIdDialog.exists());
+
+        try {
+            // get the text on the device id dialog box
+            final String[] deviceIdDialogText = deviceIdDialog.getText().split(":");
+            // look for the device id if present
+            if (deviceIdDialogText[0].equalsIgnoreCase("DeviceId")) {
+                return deviceIdDialogText[1];
+            } else {
+                return null;
+            }
+        } catch (UiObjectNotFoundException e) {
+            throw new AssertionError(e);
+        } finally {
+            // dismiss the dialog
+            UiAutomatorUtils.handleButtonClick("android:id/button1");
+        }
+    }
+
+    @Override
+    public void enableBrowserAccess() {
+        launch();
+
+        if (shouldHandleFirstRun) {
+            handleFirstRun();
+        }
+
+        // Click enable browser access
+        UiAutomatorUtils.handleButtonClick(
+                "com.microsoft.identity.testuserapp:id/buttonInstallCert"
+        );
+
+        final UiDevice device =
+                UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+
+        // Install cert
+        final UiObject certInstaller = device.findObject(new UiSelector().packageName("com.android.certinstaller"));
+        certInstaller.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
+        Assert.assertTrue(
+                "Microsoft Authenticator - cert installer dialog appears.",
+                certInstaller.exists()
+        );
+
+        UiAutomatorUtils.handleButtonClick("android:id/button1");
+    }
+
+    @Override
+    public void createPowerLiftIncident() {
+        throw new UnsupportedOperationException("Not supported on Broker Host App :(");
```

Comment:
Nit:  Could we fix that?  I mean, it's our test app, so I suppose it can do whatever we want including buffering all the logs internally and displaying them, so it might not be worthwhile.

Reply:
We could may be consume the powerlift sdk in the broker host app....we should talk about this in the next team meeting.